static/src/javascripts/projects/commercial/modules/liveblog-adverts.js

```diff
@@ -80,16 +80,24 @@ const insertAds = (paras) => {
 				classes: 'liveblog-inline',
 			});
 			// insert the ad slot container into the DOM
-			para.parentNode.insertBefore(adSlot, para.nextSibling);
+			const result = fastdom.mutate(() => {
+				para.parentNode.insertBefore(adSlot, para.nextSibling);
+			});
+			fastdomPromises.push(result);
 			// load and display the advert via GAM
 			addSlot(adSlot, false);
 			AD_COUNTER += 1;
 		}
 	}
+	return Promise.all(fastdomPromises);
 };
 
 const fill = (rules) =>
 	spaceFiller.fillSpace(rules, insertAds).then((result) => {
+		// Before the refactor of fillSpace in https://github.com/guardian/frontend/pull/24599,
+		// since insertAds returned void, result is only ever undefined and the code path below was dead.
+		// TODO investigate the impact of removing the following line.
+		result = undefined;
 		if (result && AD_COUNTER < MAX_ADS) {
```

Comment:
Wow, good find! The fact it never reaches `startListening()` again seems bad...

Reply:
Yeah if I understand right, as more blocks are added to a liveblog, we'll never insert additional ad slots.
Seems like another candidate for an experiment.